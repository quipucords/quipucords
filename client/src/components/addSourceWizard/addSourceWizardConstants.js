import React from 'react';
import AddSourceWizardStepOne from './addSourceWizardStepOne';
import AddSourceWizardStepTwo from './addSourceWizardStepTwo';
import AddSourceWizardStepThree from './addSourceWizardStepThree';

const addSourceWizardSteps = [
  {
    step: 1,
    label: '1',
    title: 'Type',
    page: <AddSourceWizardStepOne />,
    subSteps: []
  },
  {
    step: 2,
    label: '2',
    title: 'Credentials',
    page: <AddSourceWizardStepTwo />,
    subSteps: []
  },
  {
    step: 3,
    label: '3',
    title: 'Results',
    page: <AddSourceWizardStepThree />,
    subSteps: []
  }
];

const editSourceWizardSteps = [
  {
    step: 2,
    label: '1',
    title: 'Credentials',
    page: <AddSourceWizardStepTwo />,
    subSteps: []
  },
  {
    step: 3,
    label: '2',
    title: 'Results',
    page: <AddSourceWizardStepThree />,
    subSteps: []
  }
];

export { addSourceWizardSteps, editSourceWizardSteps };
