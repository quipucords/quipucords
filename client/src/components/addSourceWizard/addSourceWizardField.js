import React from 'react';
import PropTypes from 'prop-types';
import { Form, Grid } from 'patternfly-react';
import helpers from '../../common/helpers';

const AddSourceWizardField = ({ children, colLabel, colField, id, label, error, errorMessage, ...props }) => {
  const setId = id || helpers.generateId();

  return (
    <Form.FormGroup controlId={setId} validationState={error ? 'error' : null} {...props}>
      <Grid.Col componentClass={Form.ControlLabel} sm={colLabel}>
        {label}
      </Grid.Col>
      <Grid.Col sm={colField}>
        {children}
        {error && <Form.HelpBlock>{errorMessage}</Form.HelpBlock>}
      </Grid.Col>
    </Form.FormGroup>
  );
};

AddSourceWizardField.propTypes = {
  children: PropTypes.node.isRequired,
  colLabel: PropTypes.number,
  colField: PropTypes.number,
  id: PropTypes.string,
  label: PropTypes.node,
  error: PropTypes.string,
  errorMessage: PropTypes.string
};

AddSourceWizardField.defaultProps = {
  colLabel: 3,
  colField: 9,
  id: null,
  label: null,
  error: null,
  errorMessage: null
};

export { AddSourceWizardField as default, AddSourceWizardField };
