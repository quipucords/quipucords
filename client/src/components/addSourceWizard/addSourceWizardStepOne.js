import React from 'react';
import PropTypes from 'prop-types';
import { Form, Radio } from 'patternfly-react';
import { connect, store, reduxSelectors, reduxTypes } from '../../redux';
import { FormState } from '../formState/formState';
import apiTypes from '../../constants/apiConstants';

class AddSourceWizardStepOne extends React.Component {
  isStepValid = ({ values }) => {
    store.dispatch({
      type: reduxTypes.sources.VALID_SOURCE_WIZARD_STEPONE,
      source: {
        [apiTypes.API_SUBMIT_SOURCE_SOURCE_TYPE]: values.sourceType
      }
    });
  };

  render() {
    const { type } = this.props;

    return (
      <FormState validateOnMount setValues={{ sourceType: type }} validate={this.isStepValid}>
        {({ values, handleOnEvent, handleOnSubmit }) => (
          <Form horizontal onSubmit={handleOnSubmit}>
            <h3 className="right-aligned_basic-form">Select source type</h3>
            <Form.FormGroup>
              <Radio
                name="sourceType"
                value="network"
                checked={values.sourceType === 'network'}
                onChange={handleOnEvent}
              >
                Network Range
              </Radio>
              <Radio
                name="sourceType"
                value="satellite"
                checked={values.sourceType === 'satellite'}
                onChange={handleOnEvent}
              >
                Satellite
              </Radio>
              <Radio
                name="sourceType"
                value="vcenter"
                checked={values.sourceType === 'vcenter'}
                onChange={handleOnEvent}
              >
                vCenter Server
              </Radio>
            </Form.FormGroup>
          </Form>
        )}
      </FormState>
    );
  }
}

AddSourceWizardStepOne.propTypes = {
  type: PropTypes.string
};

AddSourceWizardStepOne.defaultProps = {
  type: 'network'
};

const makeMapStateToProps = () => {
  const mapSource = reduxSelectors.sources.makeSourceDetail();

  return (state, props) => ({
    ...mapSource(state, props)
  });
};

const ConnectedAddSourceWizardStepOne = connect(makeMapStateToProps)(AddSourceWizardStepOne);

export { ConnectedAddSourceWizardStepOne as default, ConnectedAddSourceWizardStepOne, AddSourceWizardStepOne };
