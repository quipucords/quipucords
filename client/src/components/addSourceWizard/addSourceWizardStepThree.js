import React from 'react';
import PropTypes from 'prop-types';
import { Icon, Spinner } from 'patternfly-react';
import { connect, reduxSelectors } from '../../redux';

const AddSourceWizardStepThree = ({ add, error, fulfilled, pending, name }) => (
  <React.Fragment>
    {error && (
      <div className="wizard-pf-complete blank-slate-pf">
        <div className="wizard-pf-success-icon">
          <Icon type="pf" name="error-circle-o" />
        </div>
        <h3 className="blank-slate-pf-main-action">Error {add ? 'Creating' : 'Updating'} Source</h3>
        <p className="blank-slate-pf-secondary-action">There are errors on the previous step</p>
      </div>
    )}
    {fulfilled && (
      <div className="wizard-pf-complete blank-slate-pf">
        <div className="wizard-pf-success-icon">
          <Icon type="pf" name="ok" />
        </div>
        <h3 className="blank-slate-pf-main-action">
          <strong>{name}</strong> was {add ? 'created' : 'updated'}.
        </h3>
      </div>
    )}
    {pending && (
      <div className="wizard-pf-process blank-slate-pf">
        <Spinner loading size="lg" className="blank-slate-pf-icon" />
        <h3 className="blank-slate-pf-main-action">{add ? 'Creating' : 'Updating'} Source...</h3>
        <p className="blank-slate-pf-secondary-action">
          Please wait while source <strong>{name}</strong> is being {add ? 'created' : 'updated'}.
        </p>
      </div>
    )}
  </React.Fragment>
);

AddSourceWizardStepThree.propTypes = {
  add: PropTypes.bool,
  error: PropTypes.bool,
  fulfilled: PropTypes.bool,
  pending: PropTypes.bool,
  name: PropTypes.string
};

AddSourceWizardStepThree.defaultProps = {
  add: false,
  error: false,
  fulfilled: false,
  pending: false,
  name: null
};

const makeMapStateToProps = () => {
  const mapSource = reduxSelectors.sources.makeSourceDetail();

  return (state, props) => ({
    ...state.addSourceWizard,
    ...mapSource(state, props)
  });
};

const ConnectedAddSourceWizardStepThree = connect(makeMapStateToProps)(AddSourceWizardStepThree);

export { ConnectedAddSourceWizardStepThree as default, ConnectedAddSourceWizardStepThree, AddSourceWizardStepThree };
