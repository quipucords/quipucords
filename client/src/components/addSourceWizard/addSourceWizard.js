import React from 'react';
import PropTypes from 'prop-types';
import { Button, Icon, Wizard } from 'patternfly-react';
import { connect, reduxActions, reduxTypes, store } from '../../redux';
import { addSourceWizardSteps, editSourceWizardSteps } from './addSourceWizardConstants';
import helpers from '../../common/helpers';
import apiTypes from '../../constants/apiConstants';

class AddSourceWizard extends React.Component {
  state = {
    activeStepIndex: 0
  };

  onCancel = () => {
    const { edit, errorStatus, fulfilled, pending } = this.props;

    const closeWizard = () => {
      this.setState({ activeStepIndex: 0 }, () => {
        store.dispatch({
          type: reduxTypes.confirmationModal.CONFIRMATION_MODAL_HIDE
        });

        store.dispatch({
          type: reduxTypes.sources.UPDATE_SOURCE_HIDE
        });

        if (fulfilled) {
          store.dispatch({
            type: reduxTypes.sources.UPDATE_SOURCES
          });
        }
      });
    };

    if (fulfilled || errorStatus >= 500 || errorStatus === 0) {
      closeWizard();
    } else if (pending) {
      store.dispatch({
        type: reduxTypes.confirmationModal.CONFIRMATION_MODAL_SHOW,
        title: `Exit Wizard`,
        heading: 'Are you sure you want to exit this wizard?',
        body: `The wizard is in a pending state and will continue ${edit ? 'updating' : 'adding'} this source.`,
        cancelButtonText: 'No',
        confirmButtonText: 'Yes',
        onConfirm: closeWizard
      });
    } else {
      store.dispatch({
        type: reduxTypes.confirmationModal.CONFIRMATION_MODAL_SHOW,
        title: 'Cancel Add Source',
        heading: `Are you sure you want to cancel ${edit ? 'updating' : 'adding'} this source?`,
        cancelButtonText: 'No',
        confirmButtonText: 'Yes',
        onConfirm: closeWizard
      });
    }
  };

  onNext = () => {
    const { activeStepIndex } = this.state;
    const { addSteps, edit, editSteps } = this.props;
    const wizardStepsLength = edit ? editSteps.length : addSteps.length;

    if (activeStepIndex < wizardStepsLength - 1) {
      this.setState({ activeStepIndex: activeStepIndex + 1 });
    }
  };

  onBack = () => {
    const { activeStepIndex } = this.state;

    if (activeStepIndex >= 1) {
      this.setState({ activeStepIndex: activeStepIndex - 1 });
    }
  };

  onSubmit = () => {
    const { addSource, edit, source, stepOneValid, stepTwoValid, updateSource } = this.props;
    const { activeStepIndex } = this.state;

    if (stepOneValid && stepTwoValid) {
      this.setState({ activeStepIndex: activeStepIndex + 1 }, () => {
        let addUpdateSourcePromise;

        if (edit) {
          addUpdateSourcePromise = updateSource(source[apiTypes.API_SUBMIT_SOURCE_ID], source);
        } else {
          addUpdateSourcePromise = addSource(source, { scan: true });
        }

        addUpdateSourcePromise.then(
          () => {
            const { props } = this;

            if (!props.show) {
              store.dispatch({
                type: reduxTypes.toastNotifications.TOAST_ADD,
                alertType: 'success',
                message: `Source ${source[apiTypes.API_SUBMIT_SOURCE_NAME]} was ${(props.edit && 'updated') ||
                  'created'}`
              });
            }
          },
          () => {
            const { props } = this;

            if (!props.show) {
              store.dispatch({
                type: reduxTypes.toastNotifications.TOAST_ADD,
                alertType: 'error',
                header: `Error ${(props.edit && 'updating') || 'creating'} source`,
                message: props.errorMessage
              });
            }
          }
        );
      });
    }
  };

  onStep = () => {
    // ToDo: wizard step map/breadcrumb/trail click, or leave disabled
  };

  renderWizardSteps() {
    const { activeStepIndex } = this.state;
    const { addSteps, edit, editSteps } = this.props;
    const wizardSteps = edit ? editSteps : addSteps;
    const activeStep = wizardSteps[activeStepIndex];

    return wizardSteps.map((step, stepIndex) => (
      <Wizard.Step
        key={step.title}
        stepIndex={stepIndex}
        step={step.step}
        label={step.label}
        title={step.title}
        activeStep={activeStep && activeStep.step}
      />
    ));
  }

  render() {
    const {
      addSteps,
      edit,
      editSteps,
      error,
      errorStatus,
      fulfilled,
      pending,
      show,
      stepOneValid,
      stepTwoValid
    } = this.props;
    const { activeStepIndex } = this.state;
    const wizardSteps = edit ? editSteps : addSteps;

    if (!show) {
      return null;
    }

    return (
      <Wizard show={show}>
        <Wizard.Header onClose={this.onCancel} title={edit ? 'Edit Source' : 'Add Source'} />
        <Wizard.Body>
          <Wizard.Steps steps={this.renderWizardSteps()} />
          <Wizard.Row>
            <Wizard.Main>
              {wizardSteps.map((step, stepIndex) => (
                <Wizard.Contents key={step.title} stepIndex={stepIndex} activeStepIndex={activeStepIndex}>
                  {wizardSteps[stepIndex].page}
                </Wizard.Contents>
              ))}
            </Wizard.Main>
          </Wizard.Row>
        </Wizard.Body>
        <Wizard.Footer>
          <Button bsStyle="default" className="btn-cancel" disabled={fulfilled} onClick={this.onCancel}>
            Cancel
          </Button>
          <Button
            bsStyle="default"
            disabled={activeStepIndex === 0 || errorStatus >= 500 || errorStatus === 0 || pending || fulfilled}
            onClick={this.onBack}
          >
            <Icon type="fa" name="angle-left" />
            Back
          </Button>
          {activeStepIndex < wizardSteps.length - 2 && (
            <Button bsStyle="primary" disabled={!stepOneValid} onClick={this.onNext}>
              Next
              <Icon type="fa" name="angle-right" />
            </Button>
          )}
          {activeStepIndex === wizardSteps.length - 2 && (
            <Button bsStyle="primary" disabled={!stepTwoValid || pending} onClick={this.onSubmit}>
              Save
              <Icon type="fa" name="angle-right" />
            </Button>
          )}
          {activeStepIndex === wizardSteps.length - 1 && (
            <Button bsStyle="primary" disabled={error || pending} onClick={this.onCancel}>
              Close
            </Button>
          )}
        </Wizard.Footer>
      </Wizard>
    );
  }
}

AddSourceWizard.propTypes = {
  addSource: PropTypes.func,
  addSteps: PropTypes.array,
  updateSource: PropTypes.func,
  show: PropTypes.bool.isRequired,
  edit: PropTypes.bool,
  editSteps: PropTypes.array,
  error: PropTypes.bool,
  errorMessage: PropTypes.string, // eslint-disable-line
  errorStatus: PropTypes.number,
  fulfilled: PropTypes.bool,
  pending: PropTypes.bool,
  source: PropTypes.object,
  stepOneValid: PropTypes.bool,
  stepTwoValid: PropTypes.bool
};

AddSourceWizard.defaultProps = {
  addSource: helpers.noop,
  addSteps: addSourceWizardSteps,
  updateSource: helpers.noop,
  edit: false,
  editSteps: editSourceWizardSteps,
  error: false,
  errorMessage: null,
  errorStatus: null,
  fulfilled: false,
  pending: false,
  source: {},
  stepOneValid: false,
  stepTwoValid: false
};

const mapDispatchToProps = dispatch => ({
  addSource: (data, query) => dispatch(reduxActions.sources.addSource(data, query)),
  updateSource: (id, data) => dispatch(reduxActions.sources.updateSource(id, data))
});

const mapStateToProps = state => ({ ...state.addSourceWizard });

const ConnectedAddSourceWizard = connect(
  mapStateToProps,
  mapDispatchToProps
)(AddSourceWizard);

export { ConnectedAddSourceWizard as default, ConnectedAddSourceWizard, AddSourceWizard };
