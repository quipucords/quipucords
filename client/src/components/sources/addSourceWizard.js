import React from 'react';
import PropTypes from 'prop-types';
import { Modal, Button, Icon, Wizard } from 'patternfly-react';
import { noop, bindMethods } from '../../common/helpers';
import { AddSourceWizardSteps } from './addSourceWizardConstants';

class AddSourceWizard extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      activeStepIndex: props.initialStepIndex || 0
    };

    bindMethods(this, [
      'confirm',
      'onStepClick',
      'onNextButtonClick',
      'onBackButtonClick'
    ]);
  }

  componentWillReceiveProps(nextProps) {}

  onStepClick(stepIndex) {
    if (stepIndex === this.state.activeStepIndex) {
      return;
    }

    this.setState({
      activeStepIndex: stepIndex
    });
  }

  onNextButtonClick() {}

  onBackButtonClick() {}

  confirm() {}

  renderWizardSteps(wizardSteps, activeStepIndex, onStepClick) {
    const activeStep = wizardSteps[activeStepIndex];

    return wizardSteps.map((step, stepIndex) => {
      return (
        <Wizard.Step
          key={stepIndex}
          stepIndex={stepIndex}
          step={step.step}
          label={step.label}
          title={step.title}
          activeStep={activeStep && activeStep.step}
          onClick={onStepClick || noop}
        />
      );
    });
  }

  renderWizardContent(wizardSteps, activeStepIndex) {
    return wizardSteps.map((step, stepIndex) => {
      let stepContent;

      switch (stepIndex) {
        case 0:
          stepContent = <div className="form-group required">Step One</div>;
          break;
        case 1:
          stepContent = <div className="form-group required">Step Two</div>;
          break;
        case 2:
          stepContent = <div className="form-group required">Step Three</div>;
          break;
        default:
          stepContent = null;
          break;
      }

      return (
        <Wizard.Contents
          key={step.title}
          stepIndex={stepIndex}
          activeStepIndex={activeStepIndex}
        >
          {stepContent}
        </Wizard.Contents>
      );
    });
  }

  render() {
    const { show, onCancel } = this.props;
    const { activeStepIndex } = this.state;

    return (
      <form className="form-horizontal">
        <Modal
          show={show}
          onHide={onCancel}
          dialogClassName="modal-dialog modal-lg wizard-pf"
        >
          <Wizard>
            <Modal.Header>
              <button
                className="close"
                onClick={onCancel}
                aria-hidden="true"
                aria-label="Close"
              >
                <Icon type="pf" name="close" />
              </button>
              <Modal.Title>Add Source</Modal.Title>
            </Modal.Header>
            <Modal.Body className="wizard-pf-body clearfix">
              <Wizard.Steps
                steps={this.renderWizardSteps(
                  AddSourceWizardSteps,
                  activeStepIndex,
                  this.onStepClick
                )}
              />
              <Wizard.Row>
                <Wizard.Main>
                  {this.renderWizardContent(
                    AddSourceWizardSteps,
                    activeStepIndex
                  )}
                </Wizard.Main>
              </Wizard.Row>
            </Modal.Body>
            <Modal.Footer className="wizard-pf-footer">
              <Button
                bsStyle="default"
                className="btn-cancel"
                onClick={onCancel}
              >
                Cancel
              </Button>
              <Button bsStyle="default" disabled>
                <Icon type="fa" name="angle-left" />Back
              </Button>
              <Button bsStyle="primary" disabled>
                Next<Icon type="fa" name="angle-right" />
              </Button>
            </Modal.Footer>
          </Wizard>
        </Modal>
      </form>
    );
  }
}

AddSourceWizard.propTypes = {
  show: PropTypes.bool.isRequired,
  initialStepIndex: PropTypes.number,
  onCancel: PropTypes.func
};

export { AddSourceWizard };
