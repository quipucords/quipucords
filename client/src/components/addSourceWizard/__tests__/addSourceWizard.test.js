import React from 'react';
import configureMockStore from 'redux-mock-store';
import { mount, shallow } from 'enzyme';
import { ConnectedAddSourceWizard, AddSourceWizard } from '../addSourceWizard';

describe('AddSourceWizard Component', () => {
  const generateEmptyStore = (obj = {}) => configureMockStore()(obj);

  it('should render a connected component', () => {
    const store = generateEmptyStore({ addSourceWizard: { show: true } });
    const component = shallow(<ConnectedAddSourceWizard />, { context: { store } });

    expect(component.dive()).toMatchSnapshot('connected');
  });

  it('should display update steps', () => {
    const store = generateEmptyStore({ addSourceWizard: { show: true, edit: true } });
    const component = shallow(<ConnectedAddSourceWizard />, { context: { store } });

    expect(component.dive()).toMatchSnapshot('connected update');
  });

  it('should not display a wizard', () => {
    const props = {
      show: false
    };

    const component = mount(<AddSourceWizard {...props} />);
    expect(component.render()).toMatchSnapshot();
  });

  it('should have specific events defined', () => {
    const props = {
      show: false
    };

    const component = mount(<AddSourceWizard {...props} />);
    const componentInstance = component.instance();

    expect(componentInstance.onCancel).toBeDefined();
    expect(componentInstance.onNext).toBeDefined();
    expect(componentInstance.onBack).toBeDefined();
    expect(componentInstance.onSubmit).toBeDefined();
    expect(componentInstance.onStep).toBeDefined();
  });
});
