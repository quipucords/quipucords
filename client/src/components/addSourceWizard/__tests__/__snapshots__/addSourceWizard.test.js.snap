// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSourceWizard Component should not display a wizard 1`] = `null`;

exports[`connected 1`] = `
<Wizard
  className=""
  dialogClassName=""
  onClose={[Function]}
  onExited={[Function]}
  show={true}
>
  <WizardHeader
    onClose={[Function]}
    title="Add Source"
  />
  <WizardBody
    className=""
  >
    <WizardSteps
      className=""
      steps={
        Array [
          <WizardStep
            activeStep={1}
            className=""
            label="1"
            onClick={[Function]}
            step={1}
            stepIndex={0}
            title="Type"
          />,
          <WizardStep
            activeStep={1}
            className=""
            label="2"
            onClick={[Function]}
            step={2}
            stepIndex={1}
            title="Credentials"
          />,
          <WizardStep
            activeStep={1}
            className=""
            label="3"
            onClick={[Function]}
            step={3}
            stepIndex={2}
            title="Results"
          />,
        ]
      }
    />
    <WizardRow
      className=""
    >
      <WizardMain
        className=""
      >
        <WizardContents
          activeStepIndex={0}
          activeSubStepIndex={null}
          className=""
          key="Type"
          stepIndex={0}
          subStepIndex={null}
        >
          <Connect(AddSourceWizardStepOne) />
        </WizardContents>
        <WizardContents
          activeStepIndex={0}
          activeSubStepIndex={null}
          className=""
          key="Credentials"
          stepIndex={1}
          subStepIndex={null}
        >
          <Connect(AddSourceWizardStepTwo) />
        </WizardContents>
        <WizardContents
          activeStepIndex={0}
          activeSubStepIndex={null}
          className=""
          key="Results"
          stepIndex={2}
          subStepIndex={null}
        >
          <Connect(AddSourceWizardStepThree) />
        </WizardContents>
      </WizardMain>
    </WizardRow>
  </WizardBody>
  <WizardFooter
    className=""
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="btn-cancel"
      disabled={false}
      onClick={[Function]}
    >
      Cancel
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={true}
      onClick={[Function]}
    >
      <Icon
        name="angle-left"
        type="fa"
      />
      Back
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="primary"
      disabled={true}
      onClick={[Function]}
    >
      Next
      <Icon
        name="angle-right"
        type="fa"
      />
    </Button>
  </WizardFooter>
</Wizard>
`;

exports[`connected update 1`] = `
<Wizard
  className=""
  dialogClassName=""
  onClose={[Function]}
  onExited={[Function]}
  show={true}
>
  <WizardHeader
    onClose={[Function]}
    title="Edit Source"
  />
  <WizardBody
    className=""
  >
    <WizardSteps
      className=""
      steps={
        Array [
          <WizardStep
            activeStep={2}
            className=""
            label="1"
            onClick={[Function]}
            step={2}
            stepIndex={0}
            title="Credentials"
          />,
          <WizardStep
            activeStep={2}
            className=""
            label="2"
            onClick={[Function]}
            step={3}
            stepIndex={1}
            title="Results"
          />,
        ]
      }
    />
    <WizardRow
      className=""
    >
      <WizardMain
        className=""
      >
        <WizardContents
          activeStepIndex={0}
          activeSubStepIndex={null}
          className=""
          key="Credentials"
          stepIndex={0}
          subStepIndex={null}
        >
          <Connect(AddSourceWizardStepTwo) />
        </WizardContents>
        <WizardContents
          activeStepIndex={0}
          activeSubStepIndex={null}
          className=""
          key="Results"
          stepIndex={1}
          subStepIndex={null}
        >
          <Connect(AddSourceWizardStepThree) />
        </WizardContents>
      </WizardMain>
    </WizardRow>
  </WizardBody>
  <WizardFooter
    className=""
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="btn-cancel"
      disabled={false}
      onClick={[Function]}
    >
      Cancel
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={true}
      onClick={[Function]}
    >
      <Icon
        name="angle-left"
        type="fa"
      />
      Back
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="primary"
      disabled={true}
      onClick={[Function]}
    >
      Save
      <Icon
        name="angle-right"
        type="fa"
      />
    </Button>
  </WizardFooter>
</Wizard>
`;
